### PR TITLE
Databases week02 homework

### PR DESCRIPTION
**Pull request by Laurelinduilwen**
part 1 is not finished, only included the statements because I ended up wasting too much time on the app

part 2 is in trigger-on-language-count.sql

part 3 is actually the entity relationship diagram I was trying to use for my week03 homework, I have used dbdiagram.io for creating it

### CODE REVIEW

File: Week2/trigger-on-language-count.sql [hunk 1]
```diff
@@ -0,0 +1,15 @@
+DELIMITER $$
+	CREATE TRIGGER language_counter
```
Comment (mogaleaf) [inline Week2/trigger-on-language-count.sql]: Good for the trigger

File: Week2/week02.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+const execQuery = util.promisify(connection.query.bind(connection));
+
+connection.connect();
+
+const queries = [
+  'SELECT name FROM city WHERE id IN(SELECT capital FROM country WHERE name = ?);',
+  'SELECT distinct language FROM countrylanguage WHERE countrycode IN (SELECT code FROM country WHERE region = ?);',
+  'SELECT COUNT(name) AS "total cities" FROM city WHERE countrycode IN(SELECT countrycode FROM countrylanguage WHERE language = ?);',
+  'SELECT name FROM country WHERE region = ? AND code IN(SELECT countrycode FROM countrylanguage WHERE language = ? AND isofficial = "T");',
+  'SELECT name FROM country WHERE region = ? AND code IN(SELECT countrycode FROM countrylanguage WHERE language = ? AND isofficial = "T");',
```
Comment (mogaleaf) [inline Week2/week02.js]: This line is the same as above ?

**Comment by Laurelinduilwen** [comment]
thank you for the feedback <3 

For the reminder on the ER diagram I actually am kind of clueless because I wasnt able to grasp what "reminder" means as a concept in this homework

a trigger on the database? setting a date as a reminder? sending notifications? I was too confused about it so I left it for later :(

File: Week2/week02.js (hunk 2 above)
Comment (Laurelinduilwen) [reply Week2/week02.js]: yep :D that is my last minute hackjob, I was trying to create separate endpoint on node and it got complicated super fast so I just copied and pasted everything in one array and kinda slipped on the last one :D Will add the final one asap

As a gaming nerd I have a habit of keeping my fingers on (W, A, S, D) as some some sort of "combat stance" , after starting to code that stance has shifted to ( Ctrl, S, X, C, V), generally when I wear down keyboards it is always the (A) button first but now my (S) button is approaching the end of its lifespan on a brand new laptop keyboard :D

File: Week2/trigger-on-language-count.sql (hunk 1 above)
Comment (Laurelinduilwen) [reply Week2/trigger-on-language-count.sql]: <3

**Comment by mogaleaf** [comment]
Consider we just ask you to store the reminder date if any, not to trigger any event on this. So basically it's just a date in the list table :)

**Comment by Laurelinduilwen** [comment]
thank you for the feedback again :) I have fixed the mistakes and implemented the feedback ^_^

File: Week2/week02.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+connection.connect();
+
+async function submitQueries() {
+  const queries = [
+    'SELECT name FROM city WHERE id IN(SELECT capital FROM country WHERE name = ?);',
+    'SELECT distinct language FROM countrylanguage WHERE countrycode IN (SELECT code FROM country WHERE region = ?);',
+    'SELECT COUNT(name) AS "total cities" FROM city WHERE countrycode IN(SELECT countrycode FROM countrylanguage WHERE language = ?);',
+    'SELECT name FROM country WHERE region = ? AND code IN(SELECT countrycode FROM countrylanguage WHERE language = ? AND isofficial = "T");',
+    'SELECT country.continent, COUNT(countrylanguage.language) AS FROM country JOIN countrylanguage ON country.code = countrylanguage.countrycode GROUP BY country.continent);',
```
Comment (mogaleaf) [inline Week2/week02.js]: Careful in America you can have multiple countries speaking english, what can you do to prevent them being count twice ?

Comment (Laurelinduilwen) [reply Week2/week02.js]: fixed, I hope this is the final solution to this :p

**Pull request closed without merging**